src/nvim/mark.c

```diff
@@ -178,6 +178,48 @@ void setpcmark(void)
     curwin->w_pcmark.lnum = 1;
   }
 
+  // When the jumplist mode is set to history, discard all subsequent 
+  // locations when setting the mark to a location that is different from the
+  // previous next position.
+  //
+  // This behavior corresponds to the following situation in a web browser.
+  // Navigate to first.com, second.com, third.com, fourth.com and then 
+  // fifth.com.  Then navigate backwards twice so that third.com is displayed.
```

Comment:
good explanation, but it should go in the user docs, not a source code comment.

Reply:
Expanded and moved into the documentation for jumplist at the bottom.